Sources/GRPCProtobuf/InputParser.swift

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import SwiftProtobuf
+import SwiftProtobufPluginLibrary
+
+import struct GRPCCodeGen.CodeGenerationRequest
+
+/// Parses a ``FileDescriptor`` object into a ``CodeGenerationRequest`` object.
+public struct InputParser {
+  public init() {}
+  public func parse(input: FileDescriptor) throws -> CodeGenerationRequest {
+    let fileName = input.name
+    let package = input.package
+    var header = String()
+    // Field number used to collect the .proto file header.
+    let headerPath = IndexPath(index: 12)
+    if let headerLocation = input.sourceCodeInfoLocation(path: headerPath) {
+      header = headerLocation.asSourceComment(
+        commentPrefix: "///",
+        leadingDetachedPrefix: "//"
+      )
+      // If the was a leading or tailing comment it won't have a blank
+      // line, after it, so ensure there is one.
+      if !header.isEmpty && !header.hasSuffix("\n\n") {
+        header.append("\n")
+      }
+    }
+    let leadingTrivia = """
+      // DO NOT EDIT.
+      // swift-format-ignore-file
+      //
+      // Generated by the Swift generator plugin for the protocol buffer compiler.
+      // Source: \(fileName)
+      //
+      // For information on using the generated types, please see the documentation:
+      //   https://github.com/apple/swift-protobuf/\n\n
+      """
+    let dependencies = input.dependencies.map {
+      CodeGenerationRequest.Dependency(module: $0.name)
+    }
```

Comment:
The dependencies should include `GRPCProtobuf` as well because that's where the serializer/deserializer are imported from.